common/src/main/java/software/tnb/common/utils/IOUtils.java

```diff
@@ -32,17 +33,30 @@ private IOUtils() {
 
     public static void writeFile(Path file, String content) {
         try {
-            Path dir = file.getParent();
-            if (!Files.exists(dir)) {
-                LOG.debug("Creating directory " + dir);
-                Files.createDirectory(dir);
-            }
+            checkIfExistAndCreate(file.getParent());
             Files.write(file, content.getBytes());
         } catch (IOException e) {
             throw new RuntimeException("Unable to write to " + file, e);
         }
     }
 
+    public static void copyFile(Path resFolder, String source, String target) throws IOException {
```

Comment:
maybe the signature can be just `void copyFile(String source, String target)` replacing the content `resFolder.resolve(target)` with just `Paths.get(target)`, then the caller will compose the target path accordingly

Reply:
hm... I don't think it would work. In target (e.g. Resource.getName()) - there is only a relative path, that makes sense only in combination with resFolder.resolve(target)

Reply:
ok

Reply:
ok